test/-ext-/debug/test_profile_frames.rb

```diff
@@ -139,6 +153,39 @@ def test_profile_frames
     }
   end
 
+  def test_profile_thread_frames
```

Comment:
As `profile_frames()` itself is throughly tested through `test_profile_frames`, I have kept tests for `rb_profile_thread_frames()` quite simple (only test that it captures frames for the specified Thread).